docs/user-guide/07-autodiff.md

```diff
@@ -5,569 +5,605 @@ permalink: /user-guide/autodiff
 
-Functions in Slang can be marked as forward-differentiable or backward-differentiable. The `fwd_diff` operator can be used on a forward-differentiable function to obtain the forward derivative propagation function. Likewise, the `bwd_diff` operator can be used on a backward-differentiable function to obtain the backward derivative propagation function. This and the next sections cover the semantics of forward and backward propagation functions, and different ways to make a function forward and backward differentiable. 
+> #### Note ####
+> Support for `IDifferentiablePtrType` is still experimental. There are no built-in types conforming to this interface, though we plan to add stdlib support in the near future.
 
-A forward derivative propagation function computes the derivative of the result value with regard to a specific set of input parameters. 
-Given an original function, the signature of its forward propagation function is determined using the following rules:
-- If the return type `R` implements `IDifferentiable` the forward propagation function will return a corresponding `DifferentialPair<R>` that consists of both the computed original result value and the (partial) derivative of the result value. Otherwise, the return type is kept unmodified as `R`.
-- If a parameter has type `T` that implements `IDifferentiable`, it will be translated into a `DifferentialPair<T>` parameter in the derivative function, where the differential component of the `DifferentialPair` holds the initial derivatives of each parameter with regard to their upstream parameters.
-- If a parameter has type `T` that implements `IDifferentiablePtrType`, it will be translated into a `DifferentialPtrPair<T>` parameter where the differential component references the differential location or buffer.
-- All parameter directions are unchanged. For example, an `out` parameter in the original function will remain an `out` parameter in the derivative function.
-- Differentiable methods cannot have a type implementing `IDifferentiablePtrType` as an `out` or `inout` parameter, or a return type. Types implementing `IDifferentiablePtrType` can only be used for input parameters to a differentiable method. Marking such a method as `[Differentiable]` will result in a compile-time diagnostic error.
+`IDifferentiablePtrType` only requires a `Differential` associated type to be specified.
 
-For example, given original function:
-```csharp
-R original(T0 p0, inout T1 p1, T2 p2, T3 p3);
-```
-Where `R`, `T0`, `T1 : IDifferentiable`, `T2` is non-differentiable, and `T3 : IDifferentiablePtrType`, the forward derivative function will have the following signature:
+#### DifferentialPtrPair<T>: Pairs of differentiable ptr types
+For types conforming to `IDifferentiablePtrType`, the corresponding pair to use for passing the derivative counterpart is `DifferentiablePtrType<T>`, which represents a pair of `T` and `T.Differential`. Objects of this type can be created using a constructor.
```

Comment:
... counterpart is `DifferentiablePtrPair<T>`.

not `DifferentiablePtrType<T>`.